### ISSUE TITLE
[VL] Unsupported type for hash: ARRAY<VARCHAR>

### ISSUE THREAD

**Issue by FelixYBW** [apache/incubator-gluten#5642]
### Backend

VL (Velox)

### Bug description

```
java.lang.RuntimeException: Exception: VeloxUserError
Error Source: USER
Error Code: INVALID_ARGUMENT
Reason: Unsupported type for hash: ARRAY<VARCHAR>
Retriable: False
Function: checkArgTypes
File: /home/binweiyang/gluten/ep/build-velox/build/velox_ep/velox/functions/sparksql/Hash.cpp
Line: 391
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 2  _ZN8facebook5velox6detail14veloxCheckFailINS0_14VeloxUserErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 3  _ZN8facebook5velox9functions8sparksql13checkArgTypesERKSt6vectorINS0_4exec17VectorFunctionArgESaIS5_EE
# 4  _ZN8facebook5velox9functions8sparksql16makeHashWithSeedERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorINS0_4exec17VectorFunctionArgESaISD_EERKNS0_4core11QueryConfigE
# 5  _ZNSt17_Function_handlerIFSt10shared_ptrIN8facebook5velox4exec14VectorFunctionEERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorINS3_17VectorFunctionArgESaISF_EERKNS2_4core11QueryConfigEEPSO_E9_M_invokeERKSt9_Any_dataSD_SJ_SN_
# 6  _ZN8facebook5velox4exec29getVectorFunctionWithMetadataERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorISt10shared_ptrIKNS0_4TypeEESaISE_EERKSA_ISB_INS0_10BaseVectorEESaISK_EERKNS0_4core11QueryConfigE
# 7  _ZN8facebook5velox4exec12_GLOBAL__N_126compileRewrittenExpressionERKSt10shared_ptrIKNS0_4core10ITypedExprEEPNS2_5ScopeERKNS4_11QueryConfigEPNS0_6memory10MemoryPoolERKSt13unordered_setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4hashISO_ESt8equal_toISO_ESaISO_EEb
# 8  _ZN8facebook5velox4exec12_GLOBAL__N_117compileExpressionERKSt10shared_ptrIKNS0_4core10ITypedExprEEPNS2_5ScopeERKNS4_11QueryConfigEPNS0_6memory10MemoryPoolERKSt13unordered_setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4hashISO_ESt8equal_toISO_ESaISO_EEb
# 9  _ZN8facebook5velox4exec18compileExpressionsERKSt6vectorISt10shared_ptrIKNS0_4core10ITypedExprEESaIS7_EEPNS4_7ExecCtxEPNS1_7ExprSetEb
# 10 _ZN8facebook5velox4exec7ExprSetC2ERKSt6vectorISt10shared_ptrIKNS0_4core10ITypedExprEESaIS8_EEPNS5_7ExecCtxEb
# 11 _ZN8facebook5velox4exec19makeExprSetFromFlagEOSt6vectorISt10shared_ptrIKNS0_4core10ITypedExprEESaIS7_EEPNS4_7ExecCtxE
# 12 _ZN8facebook5velox4exec13FilterProject10initializeEv
# 13 _ZN8facebook5velox4exec6Driver19initializeOperatorsEv
# 14 _ZN8facebook5velox4exec6Driver11runInternalERSt10shared_ptrIS2_ERS3_INS1_13BlockingStateEERS3_INS0_9RowVectorEE
# 15 _ZN8facebook5velox4exec6Driver4nextERSt10shared_ptrINS1_13BlockingStateEE
# 16 _ZN8facebook5velox4exec4Task4nextEPN5folly10SemiFutureINS3_4UnitEEE
# 17 _ZN6gluten24WholeStageResultIterator4nextEv
# 18 Java_org_apache_gluten_vectorized_ColumnarBatchOutIterator_nativeHasNext
# 19 0x00007fb687f14e28
```

It should fallback if velox doesn't support

### Spark version

None

**Comment by marin-ma** [comment]
This should be resolved by Velox PR https://github.com/facebookincubator/velox/pull/9414

**Comment by FelixYBW** [comment]
> This should be resolved by Velox PR [facebookincubator/velox#9414](https://github.com/facebookincubator/velox/pull/9414)

Oh, I thought I had seen this error before. It's there. Thank you. Let's merge the PR soon

**Comment by FelixYBW** [comment]
fixed
